sdks/python/apache_beam/examples/complete/autocomplete.py

```diff
@@ -53,7 +53,7 @@ def run(argv=None):
    | 'format' >> beam.Map(
        lambda (prefix, candidates): '%s: %s' % (prefix, candidates))
    | 'write' >> WriteToText(known_args.output))
-  p.run()
+  p.run().wait_until_finish()
```

Comment:
We do not need it, removed it.

DirectRunner uses non-daemon threads, and you are right pipeline continue to run until complete before the process exits.

DataflowRunner uses daemon threads. I kept the existing behavior and there is a comment about why it is needed. Pipelines still continue to finish tough because exiting from the runner after job submissions does not cancel the job.